backend/models/helper/model.js

```diff
@@ -1005,10 +1010,61 @@ function getMetadata(account, model, id) {
 			return Promise.reject(responseCodes.METADATA_NOT_FOUND);
 		}
 	});
+}
+
+async function getMeshById(account, model, meshId) {
+	const historyRes =  (await History.findByObjectId(account, model, meshId, {current:1}));
+	if (!historyRes) {
+		throw responseCodes.RESOURCE_NOT_FOUND;
+	}
+
+	const revisionIds = historyRes.current;
+	const projection = {
+		"parents": 1,
+		"vertices": 1,
+		"matrix": 1,
+		"faces": 1,
+		"_extRef":1
+	};
+
+	const mesh = await Scene.getObjectById(account, model, utils.stringToUUID(meshId), projection);
+	let parentMatrix = null;
+
+	if ((mesh.parents || []).length > 0) {
+		parentMatrix = await getParentMatrix(account, model, mesh.parents[0], revisionIds);
+		if (mesh.matrix) {
+			mesh.matrix = matrix.multiply(parentMatrix, mesh.matrix);
+		} else {
+			mesh.matrix = parentMatrix;
+		}
+	}
+
+	const vertices =  mesh.vertices ? new StreamBuffer({buffer: mesh.vertices.buffer, chunkSize: mesh.vertices.buffer.length}) : await Scene.getGridfsFileStream(account, model, mesh._extRef.vertices);
+	const triangles = mesh.faces ?  new StreamBuffer({buffer: mesh.faces.buffer, chunkSize: mesh.faces.buffer.length})  : await Scene.getGridfsFileStream(account, model, mesh._extRef.faces);
+
+	const combinedStream = CombinedStream.create();
+	combinedStream.append(stringToStream(["{\"matrix\":", JSON.stringify(mesh.matrix), ",\"vertices\":["].join("")));
+	combinedStream.append(vertices.pipe(new BinToVector3dStringStream({isLittleEndian: true})));
+	combinedStream.append(stringToStream("],\"triangles\":["));
+	combinedStream.append(triangles.pipe(new BinToTriangleStringStream({isLittleEndian: true})));
+	combinedStream.append(stringToStream("]}"));
+	return 	combinedStream;
+}
+
+async function getParentMatrix(account, model, parent, revisionIds) {
+	const mesh = await Scene.getBySharedId(account, model, parent, revisionIds);
```

Comment:
I think all these changes here should be in scene. Mesh/transformations are nodes within Scene so it makes sense.
We really want to get rid of model/helper/model.